MetadataExtractor/Formats/Icc/IccReader.cs

```diff
@@ -54,15 +54,17 @@ public IEnumerable<Directory> ReadJpegSegments(IEnumerable<JpegSegment> segments
                 }
             }
 
-            Directory directory = Extract(new ByteArrayReader(buffer));
+            Directory directory = Extract(buffer);
 
             ArrayPool<byte>.Shared.Return(buffer);
 
             return [directory];
         }
 
-        public IccDirectory Extract(IndexedReader reader)
+        public IccDirectory Extract(ReadOnlySpan<byte> bytes)
```

Comment:
This is slightly less flexible than before, as it requires a contiguous chunk of memory while the previous could work on a stream. Maybe that's ok. Just calling it out.